src/NuGetForUnity/Editor/Models/NugetPackageVersion.cs

```diff
@@ -244,9 +244,19 @@ public bool InRange([NotNull] NugetPackageVersion other)
 
             if (other.HasVersionRange)
             {
-                // either other or both have a version range specified. So it makes no sense to check if this is in the range of other.
-                // we can only check if the version ranges are equal.
-                return string.Equals(FullVersion, other.FullVersion, StringComparison.OrdinalIgnoreCase);
+                if (!HasVersionRange)
+                {
+                    // A 'single' version can't be in range of a version range.
```

Comment:
Can you explain this a bit more. I don't see why some version X can't be in some range? Is it because this code is in InRange function that should only be called if current package has a range?

Reply:
The function `InRange` is normally used in the following:
- version range (this) `[1.0,2.0]` InRange single version (other) `2.0`
- single version (this) `2.0` InRange single version (other) `2.0`

Now it supports:
- version range (this) `[1.0,2.0]` InRange version range (other) `[2.0,3.0]`

The if here indicates that it is used "in the wrong order" as:
- single version (this) `2.0` InRange version range (other) `[1.0,2.0]`